app/models/concerns/orchestration/dhcp.rb

```diff
@@ -171,8 +171,13 @@ def queue_dhcp_update
   # do we need to update our dhcp reservations
   def dhcp_update_required?
     # IP Address / name changed, or 'rebuild' action is triggered and DHCP record on the smart proxy is not present/identical.
-    return true if ((old.ip != ip) || (old.hostname != hostname) || provision_mac_addresses_changed? || (old.subnet != subnet) || (operatingsystem.boot_filename(old.host) != operatingsystem.boot_filename(self.host)) ||
-                    (!old.build? && build? && !all_dhcp_records_valid?))
+    return true if ((old.ip != ip) ||
+      (old.hostname != hostname) ||
+      provision_mac_addresses_changed? ||
+      (old.subnet != subnet) ||
+      (old.operatingsystem.boot_filename(old.host) != operatingsystem.boot_filename(self.host)) ||
+      ((old.host.pxe_loader == "iPXE Embedded" || self.host.pxe_loader == "iPXE Embedded") && (old.host.build != self.host.build)) ||
```

Comment:
Everywhere else we use `old.build?` and `build?`. What's the reason this uses `old.host.build` and `host.build`?

Reply:
I don't know. Not in this PR please as I smell hidden dragons :-)